lib/location.py

```diff
@@ -90,106 +90,113 @@ def geolocator(location):
 
 
 def ipcachewrite(ip_addr, location):
-    cached = os.path.join(IP2LCACHE, ip_addr)
+    """ Write a retrieved ip+location into cache
+        Can stress some filesystems after long term use, see
+        https://stackoverflow.com/questions/466521/how-many-files-can-i-put-in-a-directory
+    """
+    cachefile = os.path.join(IP2LCACHE, ip_addr)
     if not os.path.exists(IP2LCACHE):
         os.makedirs(IP2LCACHE)
-    with open(cached, 'w') as file:
-        file.write(location[0] + ';' + location[1])
+
+    with open(cachefile, 'w') as file:
+        file.write(location[0] + ';' + location[1] + ';' + location[2])
```

Comment:
I would save the whole line

Reply:
We cannot save the whole line, because this version is now a compromise between caching only ip2location format and caching all formats. Instead, the iplocating functions now return a tuple of all relevant data, and this function adapts that tuple into the line you see. It's now somewhat longer to include latlong, though